src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -551,36 +553,11 @@ bool ViewProviderSketch::keyPressed(bool pressed, int key)
     return true; // handle all other key events
 }
 
-void ViewProviderSketch::setSnapMode(SnapMode mode)
+void ViewProviderSketch::setAngleSnapping(bool enable, Base::Vector2d referencePoint)
 {
-    snapMode = mode; // to be redirected to SnapManager
-}
-
-SnapMode ViewProviderSketch::getSnapMode() const
-{
-    return snapMode; // to be redirected to SnapManager
-}
-
-void ViewProviderSketch::snapToGrid(double &x, double &y) // Paddle, when resolving this conflict, make sure to use the function in ViewProviderGridExtension
-{
-    if (snapMode == SnapMode::SnapToGrid && ShowGrid.getValue()) {
-        // Snap Tolerance in pixels
-        const double snapTol = getGridSize() / 5;
-
-        double tmpX = x, tmpY = y;
-
-        getClosestGridPoint(tmpX, tmpY);
-
-        // Check if x within snap tolerance
-        if (x < tmpX + snapTol && x > tmpX - snapTol) {
-            x = tmpX; // Snap X Mouse Position
-        }
-
-         // Check if y within snap tolerance
-        if (y < tmpY + snapTol && y > tmpY - snapTol) {
-            y = tmpY; // Snap Y Mouse Position
-        }
-    }
+    assert(snapManager);
+    snapManager->angleSnapEnabled = enable;
+    snapManager->referencePoint = referencePoint;
```

Comment:
This should be a function with the same signature as this one in SnapManager, this function delegating on that one.

This kind of direct assignment is bad for encapsulation. If you have an implementation only class (for example a PImpl structure or class), having public data members may be acceptable, as the scope is very restricted.

However, we make delegates (managers) precisely to set some basic partial architectural barriers (ViewProvider is not going to handle the snapping, so should not have access to the variables that control the implementation of snapping).

Reply:
I understand. I'll implement that